zebra/zapi_msg.c

```diff
@@ -2385,7 +2385,8 @@ static void zread_label_manager_connect(struct zserv *client,
 	if ((proto >= ZEBRA_ROUTE_MAX) || (proto <= ZEBRA_ROUTE_STATIC)) {
 		zlog_err("client %d has wrong protocol %s", client->sock,
 			 zebra_route_string(proto));
-		zsend_label_manager_connect_response(client, vrf_id, 1);
+		if (client->is_synchronous)
```

Comment:
With this check we never will send results to the upper level protocol.  It sure seems like a good thing to send the result.  Additionally synchronous just means we are waiting for a response as opposed to delaying looking for a response.  Upper level protocols should handle the failure cases when they get told about it.

Reply:
I agree with your argument. The reason is to fix current BGP label requets to the LM with a small change.

Reply:
if zebra responds immediately and the protocol is not waiting for an answer, the protocol will just read the answer the next time it wakes up from a poll session.  So I think that always sending the response is fine( in fact I've written other zapi messages that respond immediately but we do not wait for the answer )

Reply:
Ok, thank you, I'm fixing it :-)

Reply:
Donald, I've removed the fix, because the response uses the same command id "ZEBRA_LABEL_MANAGER_CONNECT", so it hurts the proxy logic: it should be "ZEBRA_LABEL_MANAGER_CONNECT_RESPONSE" or similar in order to work asynchronously. Current pushed code works. Do you want me to add "ZEBRA_LABEL_MANAGER_CONNECT_RESPONSE" command or similar, or leave it as it is? Please excuse the inconvenience.